includes/class-wcs-template-loader.php

```diff
@@ -289,4 +357,53 @@ public static function get_opc_subscription_add_to_cart() {
 		global $product;
 		wc_get_template( 'checkout/add-to-cart/simple.php', array( 'product' => $product ), '', PP_One_Page_Checkout::$template_path );
 	}
+
+	/**
+	 * Handles relocated templates.
+	 *
+	 * Hooked onto 'wc_get_templates'.
+	 *
+	 * @since 1.4.0
+	 */
+	public static function handle_relocated_templates( $template, $template_name, $args, $template_path, $default_path ) {
+		if ( ! $default_path || ! in_array( $template_name, self::$relocated_templates, true ) ) {
+			return $template;
+		}
+
+		$subscriptions_core_templates_path = WC_Subscriptions_Core_Plugin::instance()->get_subscriptions_core_directory( 'templates/' );
+		if ( $default_path === $subscriptions_core_templates_path ) {
+			return $template;
+		}
+
+		$is_default_path_wcs = false;
+		if ( class_exists( 'WC_Subscriptions_Plugin' ) ) {
+			// WCS 4+ is active.
+			$wcs_templates_path  = WC_Subscriptions_Plugin::instance()->get_plugin_directory( 'templates/' );
+			$is_default_path_wcs = $default_path === $wcs_templates_path;
+		} elseif ( class_exists( 'WC_Subscriptions' ) ) {
+			// WCS 3 or older is active.
+			$wcs_templates_path  = plugin_dir_path( WC_Subscriptions::$plugin_file ) . 'templates/';
+			$is_default_path_wcs = $default_path === $wcs_templates_path;
+		} elseif ( file_exists( dirname( $default_path ) . '/woocommerce-subscriptions.php' ) ) {
+			// WCS is installed but not active.
+			$is_default_path_wcs = true;
+		} elseif ( preg_match( '/woocommerce-subscriptions(|[A-Za-z0-9\-\_ ]+)\/templates/', $default_path ) ) {
+			$maybe_a_plugin_name = basename( dirname( $default_path ) );
+			// Avoid the case where $default_path is referring to some other plugin like woocommerce-subscriptions-extension.
+			if ( ! file_exists( dirname( $default_path ) . "/$maybe_a_plugin_name.php" ) ) {
+				$is_default_path_wcs = true;
+			}
+		}
+
+		if ( ! $is_default_path_wcs ) {
+			return $template;
+		}
+
+		return wc_get_template(
```

Comment:
Can we return `wc_get_template()` on the filter we're hooked onto?

This filter expects a string however this function returns void if I'm understanding this correctly.

Reply:
Good catch! That's a major oversight on my part.